contracts/liquidity_hub/pool-manager/src/liquidity/commands.rs

```diff
@@ -275,10 +264,43 @@ pub fn provide_liquidity(
                     amount
                 }
             }
-            PoolType::StableSwap { amp: _ } => {
-                // TODO: Handle stableswap
+            PoolType::StableSwap { amp: amp_factor } => {
+                if total_share == Uint128::zero() {
+                    // Make sure at least MINIMUM_LIQUIDITY_AMOUNT is deposited to mitigate the risk of the first
+                    // depositor preventing small liquidity providers from joining the pool
+                    let min_lp_token_amount = MINIMUM_LIQUIDITY_AMOUNT * Uint128::from(3u8);
+                    let share = Uint128::try_from(compute_d(amp_factor, &deposits).unwrap())?
+                        .checked_sub(min_lp_token_amount)
+                        .map_err(|_| {
+                            ContractError::InvalidInitialLiquidityAmount(min_lp_token_amount)
+                        })?;
+
```

Comment:
No LP tokens are being minted to the contract, i.e. the `MINIMUM_LIQUIDITY_AMOUNT`. See the case for `PoolType::ConstantProduct` when the `total_share == Uint128::zero()`

Reply:
Fixed in latest commit